### PR TITLE
Sof metadata extractor

### PR DESCRIPTION
**Pull request by ktrzcinx**
SMEX (*SOF Metadata EXtractor*) is a tool used to extract needed
information from SOF source code and output files and then put then save
them in convenient form like logs dictionary file (*ldc*).

This PR is related with https://github.com/thesofproject/sof/pull/2086

### CODE REVIEW

File: smex/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: BSD-3-Clause
```
Comment (dbaluta) [inline smex/CMakeLists.txt:1]: Please mention in the commit message why we need this tool as @mmaka1  said.

**Comment by lgirdwood** [comment]
@ktrzcinx looks like a force push is needed to kickstart the CI again too, for some reason status is pending for most of them....

File: smex/CMakeLists.txt (hunk 1 above)
Comment (ktrzcinx) [reply smex/CMakeLists.txt:1]: Added

**Comment by lgirdwood** [comment]
@xiulipan @fredoh9 Jenkins seems down ?

**Comment by xiulipan** [comment]
@lgirdwood Yes, Jenkins server has some network issue. Fixed now and run in draining mode.

**Pull request merged**